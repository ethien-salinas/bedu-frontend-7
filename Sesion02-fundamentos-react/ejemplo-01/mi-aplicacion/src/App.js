// Como fue mencionado en el archivo index.js, esta línea siempre es necesaria
// cuando un archivo contiene código de React
import React from 'react';

// Este es un componente de React que regresa un `div` con una cadena de texto
function App() {
  return (
    <div>Hola Mundo!</div>
  );
}

// Tenemos que exportar el componente para poder ser usado en cualquier otro
// archivo de React
export default App;
