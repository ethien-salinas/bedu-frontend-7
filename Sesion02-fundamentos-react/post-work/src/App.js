import React from 'react';

function App() {
  return (
    <>
      <header>
        <h1>Bienvenido a mi sitio personal</h1>
      </header>
      <main>
        <p>
          Soy un desarrollador de React que le gusta enforcarse en los estándares de la
          industria...
        </p>
      </main>
      <footer>
        <h5><a href="mailto:mi@correo.com">Sigamos en contacto!</a></h5>
        <h5><a href="https://github.com/miUsuario">Esto es mi trabajo público</a></h5>
        <h5><a href="https://twitter.com/@miUsuario">Esto pienso</a></h5>
      </footer>
    </>
  );
}

export default App;
