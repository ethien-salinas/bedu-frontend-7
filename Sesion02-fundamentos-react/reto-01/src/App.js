import React, { Component } from 'react'
import MiElemento from './MiElemento/MiElemento';

export default class App extends Component {
  render() {
    return (
      <div>
        <MiElemento />
      </div>
    )
  }
}
