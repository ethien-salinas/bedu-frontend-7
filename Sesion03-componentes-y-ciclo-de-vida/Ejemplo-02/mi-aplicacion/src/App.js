import React from 'react';

class Title extends React.Component {
  render() {
    return (
    <h2>
      {this.props.text} <br />
      {this.props.number} + {this.props.number} = {this.props.number + this.props.number}
    </h2>
    )
  }
}

class App extends React.Component {
  render() {
    return (
      <Title text="Hola Mundo" number={5}/>
    )
  }
}

export default App;
