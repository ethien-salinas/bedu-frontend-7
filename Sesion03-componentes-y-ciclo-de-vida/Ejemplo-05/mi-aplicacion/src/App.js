import React from 'react';

import Title from './Title'

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      ip: ''
    }
  }

  componentDidMount() {
    this.getData('https://checkip.amazonaws.com')
  }

  getData = async (url) => {
    // Esto solo funcionará si se tiene una extensión que mande
    // Access-Control-Allow-Origin: *
    // como respuesta del servidor
    const response = await fetch(url).then(r => r.text())

    // Todavía no vemos por qué esto funciona, se verá en la siguiente clase
    this.setState({ip: response})
  }

  render() {
    return (
      <Title ip={this.state.ip}/>
    )
  }
}

export default App;
