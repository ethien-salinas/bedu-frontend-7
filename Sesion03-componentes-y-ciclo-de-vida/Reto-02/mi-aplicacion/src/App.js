// Como fue mencionado en el archivo index.js, esta línea siempre es necesaria
// cuando un archivo contiene código de React
import React from 'react';
import Nav from './components/Nav';
import Main from './components/Main';
import Footer from './components/Footer';

import "./index.css";

// Este es un componente de React que regresa un `div` con una cadena de texto
function App() {
  return (
    <>
      <Nav />
      <Main />
      <Footer />
    </>
  );
}

// Tenemos que exportar el componente para poder ser usado en cualquier otro
// archivo de React
export default App;
