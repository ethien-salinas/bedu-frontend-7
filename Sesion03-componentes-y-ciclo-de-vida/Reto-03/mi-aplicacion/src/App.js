import React from 'react';

class App extends React.Component {
  render() {
    return (
      <p>Hola Mundo</p>
    )
  }
}

export default App;
