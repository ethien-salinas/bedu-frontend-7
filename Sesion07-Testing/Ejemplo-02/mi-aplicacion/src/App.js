import React from 'react';

function App(props) {
  // Aquí definiremos el nombre con el que conoceremos el estado y la función
  // con la que modificaremos dicho estado
  const [count, setCount] = React.useState(0)

  return (
    <>
      <button onClick={() => setCount(count + 1)}>
        Agregar uno
      </button>
      <p>La cuenta es: {count}</p>
    </>
  )
}

export default App;
