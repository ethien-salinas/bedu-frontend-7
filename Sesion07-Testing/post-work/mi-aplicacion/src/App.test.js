import React from 'react'

import App from './App'

test('The component render without crash', () => {
  const component = shallow(<App />)

  expect(component).toMatchSnapshot()
})
