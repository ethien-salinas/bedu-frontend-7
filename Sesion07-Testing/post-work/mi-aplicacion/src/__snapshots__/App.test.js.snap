// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component render without crash 1`] = `ShallowWrapper {}`;
